src/Microsoft.AspNetCore.Razor.Language/Legacy/RazorParser.cs

```diff
@@ -37,7 +37,10 @@ public virtual RazorSyntaxTree Parse(RazorSourceDocument source)
 
             var reader = new SeekableTextReader(chars, source.FileName);
 
-            var context = new ParserContext(reader, Options.DesignTimeMode);
+            var context = new ParserContext(reader, Options.DesignTimeMode)
+            {
+                StopParsingAfterFirstDirective = Options.StopParsingAfterFirstDirective
+            };
```

Comment:
Had a discussion offline and @rynowak came up with an awesomer reason to keep this on options. You can see options on the result of a parse, therefore you can identify an "only first directive" parse vs. a normal parse.